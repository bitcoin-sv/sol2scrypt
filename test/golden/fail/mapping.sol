
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.10;

contract Mapping {

    uint i = 11;
    
    mapping(address => uint) public myMap;
    
    function get(address _addr) external view returns (uint) {
        return myMap[_addr];
    }

    function get2(address _addr) public view returns (uint) {
        return myMap[_addr];
    }
    
    function set(address _addr, uint _i) external {
        myMap[_addr] = _i;
    }
}