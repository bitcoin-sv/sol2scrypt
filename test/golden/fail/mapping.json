[
    {
        "src": "test/golden/fail/mapping.sol:7:5:7:17",
        "type": "Error",
        "message": "unsupported state variable `i` with initial value"
    },
    {
        "src": "test/golden/fail/mapping.sol:15:14:15:18",
        "type": "Error",
        "message": "accessing mapping expression in non-external function `get2` is not supported"
    }
]